src/main/java/de/orchound/gameoflife/view/Painter.java

```diff
@@ -1,19 +1,30 @@
 package de.orchound.gameoflife.view;
 
 import de.orchound.gameoflife.model.Game;
+import org.joml.Vector2i;
 import org.joml.Vector2ic;
 
 public class Painter {
 
 	private final Game game;
 	private boolean currentlyPainting = false;
+	private boolean fillMode = false;
 	private boolean paintMode = false;
 
 	public Painter(Game game) {
 		this.game = game;
 	}
 
-	public void paintOrEraseCell(Vector2ic cell) {
+	public void paintAction(Vector2ic cell) {
+		if (fillMode) {
```

Comment:
The painting algorithms shouldn't be implemented in the same class.
Create a `Painter` interface and then a `TogglePainter` for the original implementation and a `FillPainter` for filling in cells.
The painters should be organized in the `Applet` and selected if an appropriate button is being clicked.